### PR TITLE
Fix validation on stale values

### PR DESCRIPTION
**Pull request by Jestermaxrko**
Fixes issue  - https://github.com/jaredpalmer/formik/issues/2083 

The cause of the issue is in this line https://github.com/jaredpalmer/formik/blob/0f960aaeeb0bdbef8312b5107cd3374884a0e62b/packages/formik/src/Formik.tsx#L185

beause of assigning value from `stateRef.current` to the `state` variable this variable contains "old" value until next re-render
https://github.com/jaredpalmer/formik/blob/0f960aaeeb0bdbef8312b5107cd3374884a0e62b/packages/formik/src/Formik.tsx#L190

And then when `setFieldValue` and `setFieldTouched` are called consecutively `stateRef.current` references to updated value imediatelly after `setFieldValue` call, but `state` variable is still references to the old value and `setFieldTouched` runs validation on that old value

Using `stateRef.current` directly gurantees that actual values are taken

### CODE REVIEW

**Review by tim-phillips — APPROVED** [review]

**Comment by ypk4** [comment]
Helpful change. Are we pending one more review to get this merged?

**Comment by pvcresin** [comment]
Thank you very much. I hope it will be released.

**Comment by quantizor** [comment]
Would love a test please

File: packages/formik/src/Formik.tsx [hunk 1]
```diff
@@ -548,7 +553,7 @@ export function useFormik<Values extends FormikValues = FormikValues>({
       const willValidate =
         shouldValidate === undefined ? validateOnBlur : shouldValidate;
       return willValidate
-        ? validateFormWithHighPriority(state.values)
```
Comment (quantizor) [inline packages/formik/src/Formik.tsx:551]: This is actually the exact same thing. If you look at L190

```
const state = stateRef.current;
```

**Comment by gone-skiing** [comment]
Any chance this can get merged and released? Seems like a helpful change to keep on the shelf.

**Comment by Jestermaxrko** [comment]
@quantizor @jaredpalmer I've updated this branch, also added test. Could you take a look ?

**Comment by Christine-Pinto** [comment]
Hey! I'm experimenting with an AI-powered QA tool that reviews PRs for edge cases and missed      
  patterns. I ran it against this PR and thought the findings might be useful.                      
                                                                                                                                       
 
## Summary                                                                                        
                                                                                                    
  The PR correctly identifies the root cause: `state.values` is captured at render time and becomes 
  stale when multiple state updates happen in the same tick. The fix to use `stateRef.current` in   
  `setFieldTouched` is the right approach.                                                          
                                                                                                    
  However, the tool found the **same stale-state pattern in other methods** that weren't updated:   
                                                                                                    
  ## Findings                                                                                       
                                                                                                    
  | Location | Line | Issue |                                                                       
  |----------|------|-------|                                                                       
  | `setFieldValue` (function updaters) | 646 | Uses `state.values` instead of                      
  `stateRef.current.values` |                                                                       
  | `validateField` | 567, 583 | Uses `state.values` for validation lookup |                        
  | `validateForm` default param | 531 | Default `values = state.values` captures stale state |     
                                                                                                    
  ## Impact                                                                                         
                                                                                                    
  If a user calls:                                                                                  
                                                                                        
```
  setFieldValue('a', prev => prev + 1);                                                             
  setFieldValue('a', prev => prev + 1); // Second update uses stale value                           
  Or:                                                                                               
  setFieldValue('email', 'invalid');                                                                
  validateField('email'); // Validates old value, returns no error          
```                    
                                                                                                    
  Suggestion                                                                                        
                                                                                                    
  Consider applying the same stateRef.current pattern to all imperative methods that access form    
  values.  
                                                                                                    
  ---🤖 This analysis was generated by an experimental QA agent I'm building. Feedback welcome!
